src/mixins/legal-api-mixin.ts

```diff
@@ -158,53 +160,135 @@ export default class LegalApiMixin extends Mixins(FilingTemplateMixin) {
   }
 
   /**
-   * Fetch business data for current business.
-   * @param datatype The type of data to request.
-   * @returns a promise to return the business base info or of the specified data type.
+   * Fetches the business info of the current business.
+   * @returns a promise to return the data
    */
-  async getBusinessData (datatype: string = null): Promise<any> {
-    if (!this.getBusinessId) throw new Error('Invalid parameter \'businessIdentifier\'')
+  async fetchBusinessInfo (): Promise<BusinessInformationIF> {
+    if (!this.getBusinessId) throw new Error('Invalid business id')
 
-    let url = `businesses/${this.getBusinessId}`
-    if (datatype) {
-      url += `/${datatype}`
-    }
+    const url = `businesses/${this.getBusinessId}`
 
     return axios.get(url)
       .then(response => {
-        if (response && response.data) {
+        if (response?.data) {
+          return response.data.business
```

Comment:
This is where we deal with converting what the API returns to the typed object we care about (in this case, BusinessInformationIF).

The other methods here are similar.